### PR TITLE
KOGITO-2816: Create `editor-api` package

### PR DESCRIPTION
**Pull request by tiagobento**
See https://issues.redhat.com/browse/KOGITO-2816

Next steps: https://issues.redhat.com/browse/KOGITO-2815

On this PR:
- Moved all interfaces that are needed by both Envelope and Channels to `@kogito-tooling/microeditor-envelope-protocol`
- Moved creation of `window.envelope` to `@kogito-tooling/kie-bc-editors`, since React-based editors will consume the services using `useEnvelopeContext`.
- Created `@kogito-tooling/editor-api` to be the only dependency of packaged providing Editors.
- Created new `getElementPosition` method on Editor interface so that al Editors can use the Guided Tour API.
- Removed `@kogito-tooling/core-api`, since all the common interfaces are now in `@kogito-tooling/microeditor-envelope-protocol`.
- Moved `EditorFactory` to `@kogito-tooling/editor-api`, since it's the job of packages providing Editor to define a factory.
- Created `@kogito-tooling/keyboard-shortcuts-channel` to provide Hooks only used in Channels.
- https://issues.redhat.com/browse/KOGITO-2496
- https://issues.redhat.com/browse/KOGITO-2573

This is a first step to a better organization of our modules.

### CODE REVIEW

**Comment by tiagobento** [comment]
cc @paulovmr @caponetto

**Review by ljmotta — APPROVED** [review]
LGTM

File: packages/microeditor-envelope-protocol/src/kogito/api/ChannelKeyboardEvent.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
+
```
Comment (caponetto) [inline packages/microeditor-envelope-protocol/src/kogito/api/ChannelKeyboardEvent.ts]:
```suggestion
```

File: packages/vscode-extension/src/KogitoEditorFactory.ts [hunk 2]
```diff
@@ -16,13 +16,14 @@
...
 import { KogitoEditor } from "./KogitoEditor";
-import { KogitoEdit, ResourceContentService, Routes } from "@kogito-tooling/core-api";
+import { KogitoEdit, ResourceContentService } from "@kogito-tooling/microeditor-envelope-protocol";
 import { VsCodeNodeResourceContentService } from "./VsCodeNodeResourceContentService";
 import { VsCodeResourceContentService } from "./VsCodeResourceContentService";
 
 import * as vscode from "vscode";
 import * as nodePath from "path";
 import { DefaultVsCodeRouter } from "./DefaultVsCodeRouter";
+import { Routes } from "@kogito-tooling/microeditor-envelope-protocol";
```
Comment (caponetto) [inline packages/vscode-extension/src/KogitoEditorFactory.ts]: Please merge this import with line 19.

**Review by caponetto — COMMENTED** [review]
LGTM! Left two minor comments :lipstick:

File: packages/microeditor-envelope-protocol/src/kogito/api/ChannelKeyboardEvent.ts (hunk 1 above)
Comment (tiagobento) [reply packages/microeditor-envelope-protocol/src/kogito/api/ChannelKeyboardEvent.ts]: Done.

File: packages/vscode-extension/src/KogitoEditorFactory.ts (hunk 2 above)
Comment (tiagobento) [reply packages/vscode-extension/src/KogitoEditorFactory.ts]: Done.

**Review by caponetto — APPROVED** [review]

**Review by ederign — APPROVED** [review]

**Pull request merged**